### PR TITLE
update pegged cbr contract

### PR DESCRIPTION
**Pull request by xiaozhou**

### CODE REVIEW

File: contracts/pegged/PeggedTokenBridge.sol [hunk 1]
```diff
@@ -50,11 +58,16 @@ contract PeggedTokenBridge {
         address _token,
         uint256 _amount,
         uint64 _nonce
-    ) external {
-        bytes32 id = keccak256(abi.encodePacked("burn", _token, msg.sender, _amount, _nonce));
-        require(records[id] == false, "record exists");
-        records[id] = true;
+    ) external whenNotPaused {
+        bytes32 burnId = keccak256(abi.encodePacked(msg.sender, _token, _amount, _nonce, uint64(block.chainid)));
```
Comment (Dominator008) [inline contracts/pegged/PeggedTokenBridge.sol:62]: Note that this can match a `mintId`, but maybe that's fine given the nonce is user-specified.

Comment (xiaozhou) [reply contracts/pegged/PeggedTokenBridge.sol:62]: mintId's hashed data length is 20+20+32+8+32 = 112, while burnId's hashed data length is 20+20+32+8+8 = 88.

File: contracts/pegged/OriginalTokenVaults.sol [hunk 2]
```diff
@@ -7,21 +7,32 @@ import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
...
-contract OriginalTokenVaults is ReentrancyGuard {
+contract OriginalTokenVaults is ReentrancyGuard, Pauser, VolumeControl, DelayedTransfer {
     using SafeERC20 for IERC20;
 
     ISigsVerifier public immutable sigsVerifier;
 
-    uint64 public mintseq; // ensure unique Mint event, start from 1
+    mapping(bytes32 => bool) public records;
+
+    // TODO: may remove valuts as records are kept in sgn
```
Comment (Dominator008) [inline contracts/pegged/OriginalTokenVaults.sol:24]: Does this still make sense now that `ref_chain_id` has multiple meanings?

Comment (xiaozhou) [reply contracts/pegged/OriginalTokenVaults.sol]: removed

**Review by Dominator008 — APPROVED** [review]

**Pull request merged**